SMLHelper/Utility/BasicText.cs

```diff
@@ -0,0 +1,435 @@
+            textText.fontSize = cloneSize ? uGUI.main.intro.mainText.text.fontSize : size;
+            textText.fontStyle = cloneStyle ? uGUI.main.intro.mainText.text.fontStyle : style;
+            textText.alignment = cloneAlign ? uGUI.main.intro.mainText.text.alignment : align;
+            textText.color = cloneColor ? uGUI.main.intro.mainText.text.color : color;
+            textText.material = cloneMaterial ? uGUI.main.intro.mainText.text.material : material;
+
+            // Sets it to SN's highest layer -- note it will not appear on top of the black "you are dead" overlay. To do that
+            // requires setting it HIGHER than uGUI.main.overlays.overlays[0].graphic  -- which will make SN throw a sloppy exception
+            // in the log, but will WORK.
+            textObject.transform.SetParent(uGUI.main.screenCanvas.transform, false); // Parents our text to the black overlay
+            textText.canvas.overrideSorting = true;              // Turn on canvas sort override so the layers will work                    
+            textObject.layer = 31;                               // Set to what seems to be Subnautica's highest layer
+        }
+
+        protected float x { get; set; } = 0;          // X position anchor
+        protected float y { get; set; } = 210f;       // Y position anchor (defaults to a comfortable centered about 1/3 from top of screen)
+        protected bool cloneAlign { get; set; }       // True if we're cloning Subnautica's "Press Any Button To Begin" alignment
+        protected bool cloneColor { get; set; }       // True if we're cloning Subnautica's "Press Any Button To Begin" color
+        protected bool cloneSize { get; set; }        // True if we're cloning Subnautica's "Press Any Button To Begin" fontsize
+        protected bool cloneFont { get; set; }        // True if we're cloning Subnautica's "Press Any Button To Begin" font
+        protected bool cloneStyle { get; set; }       // True if we're cloning Subnautica's "Press Any Button To Begin" font style
+        protected bool cloneMaterial { get; set; }    // True if we're cloning Subnautica's "Press Any Button To Begin" material
+        protected TextAnchor align { get; set; }      // text alignment
+        protected Color color { get; set; }           // text color
+        protected int size { get; set; }              // text size
+        protected Font font { get; set; }             // text font
+        protected FontStyle style { get; set; }       // text font style
+        protected Material material { get; set; }     // text material
+        protected GameObject textObject { get; set; } = null;          // Our game object
+        protected uGUI_TextFade textFade { get; set; } = null;         // Our text fader
+        protected Text textText { get; set; } = null;                  // Our text object
+        protected ContentSizeFitter textFitter { get; set; } = null;   // Our content size fitter
```

Comment:
Is there expectation that we will be extending/inheriting the `BasicText` class?

If not, these could be made `internal`

Reply:
Made them internal